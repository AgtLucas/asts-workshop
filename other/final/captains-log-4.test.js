import path from 'path'
import testBabel from '../test-babel'
import captainsLog from './captains-log-4'

testBabel({
  plugin: captainsLog,
  snapshot: true,
  fixtures: path.join(__dirname, './__fixtures__'),
  tests: [
    {
      title: 'outside a function',
      code: `console.log('sup dawg')`,
      output: `console.log('1:0', 'sup dawg');`,
      snapshot: false,
    },
    `
      function add(a, b) {
        console.log(a, b)
        return a + b
      }
    `,
    `
      const multiply = (a, b) => {
        console.log(a, b)
        return a * b
      }
    `,
    `
      const divide = function(a, b) {
        console.log(a, b)
        return a / b
      }
    `,
    {fixture: 'log.js'},
    {fixture: 'log2.js'},
  ],
})
