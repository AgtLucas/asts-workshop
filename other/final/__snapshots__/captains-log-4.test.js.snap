// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2. captains-log 1`] = `
"
function add(a, b) {
  console.log(a, b)
  return a + b
}

      ↓ ↓ ↓ ↓ ↓ ↓

function add(a, b) {
  console.log(\\"add 2:2\\", a, b);
  return a + b;
}
"
`;

exports[`3. captains-log 1`] = `
"
const multiply = (a, b) => {
  console.log(a, b)
  return a * b
}

      ↓ ↓ ↓ ↓ ↓ ↓

const multiply = (a, b) => {
  console.log(\\"multiply 2:2\\", a, b);
  return a * b;
};
"
`;

exports[`4. captains-log 1`] = `
"
const divide = function(a, b) {
  console.log(a, b)
  return a / b
}

      ↓ ↓ ↓ ↓ ↓ ↓

const divide = function (a, b) {
  console.log(\\"divide 2:2\\", a, b);
  return a / b;
};
"
`;

exports[`5. captains-log 1`] = `
"
module.exports = hello

function hello() {
  console.log('in hello here')
}

      ↓ ↓ ↓ ↓ ↓ ↓

module.exports = hello;

function hello() {
  console.log('hello 4:2', 'in hello here');
}
"
`;

exports[`6. captains-log 1`] = `
"
module.exports = helloWorld

function helloWorld() {
  console.log('in helloWorld here')
}

      ↓ ↓ ↓ ↓ ↓ ↓

module.exports = helloWorld;

function helloWorld() {
  console.log('helloWorld 4:2', 'in helloWorld here');
}
"
`;
